types/history/index.d.ts

```diff
@@ -49,7 +49,10 @@ export namespace History {
       location: Location<S>,
       action: Action,
     ) => void;
-    export type LocationState = any;
+    // The value type here is a "poor man's `unknown`". When these types support TypeScript
+    // 3.0+, we can replace this with `unknown`.
+    type PoorMansUnknown = {} | null | undefined;
```

Comment:
https://github.com/gcanti/unknown-ts/blob/master/src/index.ts#L1 maybe?

Reply:
This has worked well for me so far, I've also used it in the Node types.